### PR TITLE
Adds a RateLimiting sampler based on the Jaeger sampler for Java.

### PR DESCRIPTION
**Pull request by samsp-msft**
https://github.com/open-telemetry/opentelemetry-java/blob/main/sdk-extensions/jaeger-remote-sampler/src/main/java/io/opentelemetry/sdk/extension/trace/jaeger/sampler/RateLimitingSampler.java

Fixes #
Design discussion issue #

## Changes

Please provide a brief description of the changes here.

## Merge requirement checklist

* [x] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/main/CONTRIBUTING.md) guidelines followed (license requirements, nullable enabled, static analysis, etc.)
* [ ] Unit tests added/updated
* [ ] Appropriate `CHANGELOG.md` files updated for non-trivial changes
* [ ] Changes in public API reviewed (if applicable)

### CODE REVIEW

**Comment by cijothomas** [comment]
> [CLA Missing ID](https://api.easycla.lfx.linuxfoundation.org/v2/repository-provider/github/sign/9522512/261573555/1967/#/?version=2) [CLA Not Signed](https://api.easycla.lfx.linuxfoundation.org/v2/repository-provider/github/sign/9522512/261573555/1967/#/?version=2)
> 
> * ❌ The email address for the commit ([9d21003](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/commit/9d2100346677b306e76181951ab48dddff329731)) is not linked to the GitHub account, preventing the EasyCLA check. Consult [this Help Article](https://confluence.linuxfoundation.org/pages/viewpage.action?pageId=86641302) and [GitHub Help](https://help.github.com/en/github/committing-changes-to-your-project/why-are-my-commits-linked-to-the-wrong-user) to resolve. (To view the commit's email address, add .patch at the end of this PR page's URL.) For further assistance with EasyCLA, [please submit a support request ticket](https://jira.linuxfoundation.org/servicedesk/customer/portal/4).

This is a pre-req for reviewing/merging any PRs! Please go through the process of signing EasyCLA. (its a one time process only)

**Comment by samsp-msft** [comment]
I think I fixed the cla stuff?

**Comment by cijothomas** [comment]
> I think I fixed the cla stuff?

mm.. I dont think it worked... Can you try once more?
image

File: src/OpenTelemetry.Extensions/README.md [hunk 1]
```diff
@@ -79,3 +79,24 @@ var tracerProvider = Sdk.CreateTracerProviderBuilder()
...
 started activity.
 Do not use slow or intensive operations.
+
+
+### RateLimitingSampler
+
+The rate limiting sampler is a sampler that will limit the number of traces to the specified rate per second.
+It is typically used in conjunction with the ParentBasedSampler to ensure that the rate limiting sampler is only applied to the root spans.
+If a request comes in without a sampling decision, the rate limiting sampler will make a decision based on the rate limit.
+The sampling decision is stored in the activity context and is passed along with calls to other services so the same decision is used for the entire trace.
```
Comment (cijothomas) [inline src/OpenTelemetry.Extensions/README.md:89]: what is the decision that is stored in the activity context?

**Pull request closed without merging**